components/system-broker/internal/director/client.go

```diff
@@ -103,170 +104,124 @@ func (c *GraphQLClient) RequestPackageInstanceCredentialsCreation(ctx context.Co
-					  additionalQueryParams
-					  requestAuth {
-						csrf {
-						  tokenEndpointURL
-						}
-					  }
-					  credential {
-						... on OAuthCredentialData {
-						  clientId
-						  clientSecret
-						  url
-						}
-						... on BasicCredentialData {
-						  username
-						  password
-						}
-					  }
-					}
 					status {
 					  condition
 					  timestamp
 					  message
 					  reason
 					}
 			  	 }
-				}`, in.PackageID, inContext, input))
+				}`, in.PackageID, in.AuthID, inContext, inputParams))
 
-	var resp RequestPackageInstanceCredentialsOutput
+	var resp PackageInstanceAuthOutput
 	if err = c.gcli.Do(ctx, gqlRequest, &resp); err != nil {
 		return nil, errors.Wrap(err, "while executing GraphQL call to create package instance auth")
 	}
 
 	return &resp, nil
 }
 
-func (c *GraphQLClient) FindPackageInstanceCredentialsForContext(ctx context.Context, in *FindPackageInstanceCredentialsByContextInput) (*FindPackageInstanceCredentialsOutput, error) {
+func (c *GraphQLClient) FetchPackageInstanceCredentials(ctx context.Context, in *PackageInstanceInput) (*PackageInstanceCredentialsOutput, error) {
 	if _, err := govalidator.ValidateStruct(in); err != nil {
 		return nil, errors.Wrap(err, "while validating input")
 	}
 
-	gqlRequest := gcli.NewRequest(fmt.Sprintf(`query {
-			result: application(id: "%s") {
-						package(id: "%s") {
-							%s
-					 	}
-					}
-				}`, in.ApplicationID, in.PackageID, c.outputGraphqlizer.ForPackage()))
+	gqlRequest := gcli.NewRequest(fmt.Sprintf(`query{
+			  result:packageByInstanceAuth(authID:%q){
+				apiDefinitions{
+				  data{
+					name
+					targetURL
+				  }
+				}
+				instanceAuth(id: %q){
+				  %s
+				}
+			  }
+	}`, in.InstanceAuthID, in.InstanceAuthID, c.outputGraphqlizer.ForPackageInstanceAuth()))
 
-	var resp struct {
-		Result schema.ApplicationExt `json:"result"`
+	var response struct {
+		Package *schema.PackageExt `json:"result"`
 	}
-	err := c.gcli.Do(ctx, gqlRequest, &resp)
-	if err != nil {
-		return nil, errors.Wrap(err, "while executing GraphQL call to get package instance auths")
+	if err := c.gcli.Do(ctx, gqlRequest, &response); err != nil {
+		return nil, errors.Wrap(err, "while executing GraphQL call to get package instance auth")
 	}
-	var authsResp []*schema.PackageInstanceAuth
-	for _, auth := range resp.Result.Package.InstanceAuths {
-		if auth == nil {
-			continue
-		}
 
-		var authContext map[string]string
-		if err := json.Unmarshal([]byte(*auth.Context), &authContext); err != nil {
-			return nil, errors.Wrap(err, "while unmarshaling auth context")
-		}
-
-		shouldReturn := true
-		for key, value := range in.Context {
-			authContextValue, found := authContext[key]
-			if !found || authContextValue != value {
-				shouldReturn = false
-			}
-		}
-
-		if shouldReturn {
-			authsResp = append(authsResp, auth)
-		}
+	if response.Package == nil || response.Package.InstanceAuth == nil || response.Package.InstanceAuth.Context == nil {
```

Comment:
It can not be nil, see: https://github.com/kyma-incubator/compass/blob/system-broker-bind-api/components/system-broker/internal/director/client.go#L150